packages/http-server-csharp/src/service.ts

```diff
@@ -673,19 +674,18 @@ export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>)
         ]);
       }
       const responseInfo = this.#getOperationResponse(httpOperation);
-      let status: string = "200";
-      let response: CSharpType = new CSharpType({
-        name: "void",
-        namespace: "System",
-        isBuiltIn: true,
-        isValueType: false,
-      });
-      if (responseInfo !== undefined) {
-        [status, response] = responseInfo;
-      }
+      const status = responseInfo?.statusCode ?? 200;
+      const response: CSharpType =
+        responseInfo?.resultType ??
+        new CSharpType({
+          name: "void",
+          namespace: "System",
+          isBuiltIn: true,
+          isValueType: false,
+        });
 
       const hasResponseValue = response.name !== "void";
-      const resultString = `${status === "204" ? "NoContent" : "Ok"}`;
+      const resultString = `${status === 204 ? "NoContent" : "Ok"}`;
```

Comment:
It would be nice to have some more general support for status code here, but this is a big improvement